packages/core-app-api/src/app/TranslationResource.ts

```diff
@@ -18,13 +18,13 @@ import { TranslationRef } from '@backstage/core-plugin-api/alpha';
 
 /** @alpha */
 export type TranslationMessages<T> = T extends TranslationRef<infer R>
-  ? Record<string, Partial<R>>
+  ? Partial<R>
```

Comment:
for types, last version `createTranslationResource` with `lazyMessages` options need to be 
```ts
createTranslationResource({
    lazyMessages: {
        zh: () => Promise.resolve({ messages: { zh: { language: '语言' }}})
    }
})
```
due to `TranslationMessages ` wrap `Record` to specify language key